sgt-back-end/index.js

```diff
@@ -0,0 +1,142 @@
+const express = require("express")
+const app = express()
+
+const pg = require("pg")
+const db = new pg.Pool({
+  connectionString: "postgres://dev:lfz@localhost/studentGradeTable"
+})
+
+app.use(express.json())
+
+app.get("/api/grades",(req,res)=>{
+  const sql = `
+    select *
+    from "grades"
+  `
+
+  db.query(sql)
+    .then(result=>{
+      const grades = result.rows
+      res.status(200).json(grades)
+    })
+
+    .catch(err=>{
+      res.status(500).json({error: "An unexpected error occurred."})
+    })
+})
+
+
+app.post("/api/grades", (req, res) => {
+  if ("name" in req.body && "course" in req.body && "score" in req.body &&
+  !isNaN(req.body.score) && Number.isInteger(parseFloat(req.body.score)) &&
+  req.body.score>0 && req.body.score<=100 && Object.keys(req.body).length===3){
+
+    const sql=`
+      insert into "grades" ("name","course","score")
+      values ('${req.body.name}', '${req.body.course}', ${parseInt(req.body.score)})
```

Comment:
While this does work here, you want to be adding the values into the query as per the instructions in which you put $1, $2, etc into the query and use an array to hold the values and then pass that in as the second argument to query. The reasons for this are security and and the likelihood of making an error during the interpolation of the values